### PR TITLE
Minhphu/server

### PR DESCRIPTION
**Pull request by minhphu20**
Implement server side rendering for user page:
https://sites.google.com/codeustudents.com/spring-2019/weeks-3-7-guided-projects/server-side-rendering?authuser=1

Moreover, we now have more beautiful user page url: /users/Ada instead of /user-page.html?user=Ada

### CODE REVIEW

File: src/main/java/com/google/codeu/servlets/UserServlet.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+  @Override
+  public void init() {
+    datastore = new Datastore();
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response)
+    throws IOException, ServletException {
+
```
Comment (cindyxue) [inline src/main/java/com/google/codeu/servlets/UserServlet.java]: extra line

File: src/main/webapp/WEB-INF/user.jsp [hunk 2]
```diff
@@ -0,0 +1,108 @@
+<%@ page import="java.lang.reflect.Type" %>
+<%@ page import="com.google.gson.reflect.TypeToken" %>
+<%@ page import="com.google.gson.Gson" %>
+<%@ page import="java.util.Date" %>
+<%@ page import="java.util.List" %>
+<%@ page import="com.google.codeu.data.Message" %>
+
+<!DOCTYPE html>
+<%
```
Comment (cindyxue) [inline src/main/webapp/WEB-INF/user.jsp:10]: Add comments

File: src/main/webapp/WEB-INF/user.jsp [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+          <li><a href="/stats.html">Stats</a></li>
+          <li><a href="/map.html">Map</a></li>
+          <li><a href="/feed.html">Feed</a></li>
+          <li><a href="/chart.html">Chart</a></li>
+        </ul>
+      </nav>
+      <h1 id="page-title" class="shadow"><%= user %></h1>
+
+      <%
+      if (isUserLoggedIn) {
```
Comment (cindyxue) [inline src/main/webapp/WEB-INF/user.jsp:45]: Add comments

File: src/main/webapp/WEB-INF/user.jsp [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+          <input type="submit" value="Submit" class="rounded">
+        </form>
+        <hr/>
+      <% 
+      } 
+      %>
+
+      <div id="message-container">
+        <% 
+        String json = (String) request.getAttribute("messages");
```
Comment (cindyxue) [inline src/main/webapp/WEB-INF/user.jsp:84]: Add comments

**Review by cindyxue — CHANGES_REQUESTED** [review]

File: src/main/java/com/google/codeu/servlets/UserServlet.java (hunk 1 above)
Comment (minhphu20) [reply src/main/java/com/google/codeu/servlets/UserServlet.java]: removed

File: src/main/webapp/WEB-INF/user.jsp (hunk 2 above)
Comment (minhphu20) [reply src/main/webapp/WEB-INF/user.jsp:10]: done

File: src/main/webapp/WEB-INF/user.jsp (hunk 3 above)
Comment (minhphu20) [reply src/main/webapp/WEB-INF/user.jsp:45]: done

File: src/main/webapp/WEB-INF/user.jsp (hunk 4 above)
Comment (minhphu20) [reply src/main/webapp/WEB-INF/user.jsp:84]: done

**Review by cindyxue — APPROVED** [review]

**Pull request merged**